auto_submit/lib/requests/github_webhook.dart

```diff
@@ -25,19 +33,432 @@ class GithubWebhook {
     final String rawBody = await request.readAsString();
     final body = json.decode(rawBody) as Map<String, dynamic>;
 
-    if (!body.containsKey('pull_request') || !body['pull_request'].containsKey('labels')) {
+    if (!body.containsKey('pull_request') ||
+        !body['pull_request'].containsKey('labels')) {
       return Response.ok(jsonEncode(<String, String>{}));
     }
 
     PullRequest pullRequest = PullRequest.fromJson(body['pull_request']);
-    hasAutosubmit = pullRequest.labels!.any((label) => label.name == 'autosubmit');
+    hasAutosubmit =
+        pullRequest.labels!.any((label) => label.name == 'autosubmit');
 
     if (hasAutosubmit) {
-      // TODO(kristinbi): Check if PR can be submitted. https://github.com/flutter/flutter/issues/98707
+      // TODO(kristin): hardcode the githubToken this time for test, will add it to env later.
+      // final String githubToken = Platform.environment['AUTOSUBMIT_TOKEN']!;
+
+      final String githubToken = 'f4d8bc081a5f3ad57f3df3a99ec0417b269cec90';
+      final GithubService gitHub =
+          config.createGithubServiceWithToken(githubToken);
+
+      if (!(body.containsKey('repository') &&
+          body['repository'].containsKey('full_name') &&
+          body.containsKey('number'))) {
+        return Response.ok(jsonEncode(<String, String>{}));
+      }
+      RepositorySlug slug =
+          RepositorySlug.full(body['repository']['full_name']);
+      int number = body['number'];
+
+      // Use github Rest API to get this single pull request.
+      final PullRequest pr =
+          await gitHub.getPullRequest(slug, prNumber: number);
+      logger.info('Get the pull request $pr');
+
+      _AutoMergeQueryResult queryResult =
+          await _parseQueryData(pr, gitHub, body);
+      if (await shouldMergePullRequest(queryResult, slug, gitHub)) {
+        // TODO(Kristin): publish the pr with 'autosubmit' labek to pubsub. https://github.com/flutter/flutter/issues/98704
```

Comment:
This logic should be in the cronjob API to keep pulling pubsub queues. Here, we can simply add a note: `log.info('adding pr to pubsub')`.

Reply:
Hi Keyong, do you mean that we should move the logic of  TODO to the cronjob API? I think line 67- 69 here is usde to check wheter this pr should be merged here.

Reply:
Ideally, we should push whatever PRs that are available to the pubsub queue, and leave the check logic in the cronjob API.

Considering what @CaseyHillers suggested above, we can leave the logic here to check immediately whether a PR should merge, prior the pubsub is ready.

Reply:
Oh I see, I will leave them here prior the pubsub is ready.